src/noit_metric_tag_search.c

```diff
@@ -1373,8 +1354,27 @@ noit_metric_tag_search_evaluate_against_metric_id(const noit_metric_tag_search_a
     return mtev_false;
   }
 
-  const noit_metric_tagset_t *tagsets[3] = { &tagset_check, &tagset_stream, &tagset_measurement };
-  mtev_boolean ok = noit_metric_tag_search_evaluate_against_tags_multi(search, tagsets, 3);
+  // setup implicit tags
+  noit_metric_tagset_t tagset_implicit;
+  MKTAGSETCOPY(tagset_implicit);
+  char implicit_str[NOIT_IMPLICIT_TAG_MAX_PAIR_LEN + 1 +
+                    sizeof("__check_uuid:") + UUID_PRINTABLE_STRING_LENGTH];
+  size_t uuid_index =
+      snprintf(implicit_str, sizeof(implicit_str),
+               "__name:%.*s,__check_uuid:", id->name_len, id->name);
+  mtev_uuid_unparse_lower(id->id, implicit_str + uuid_index);
+  noit_metric_add_implicit_tags_to_tagset(
+      implicit_str, uuid_index + UUID_PRINTABLE_STRING_LENGTH, &tagset_implicit,
+      NULL);
+  // TODO: should the following use NOIT_METRIC_TAGSET_STREAM or NOIT_METRIC_TAGSET_CHECK?
```

Comment:
Does anyone know what `noit_metric_tagset_fixup_hook_invoke` does? Should it be passed `NOIT_METRIC_TAGSET_STREAM` or `NOIT_METRIC_TAGSET_CHECK` when being used for the implicit tag tagset containing `__name` and `__check_uuid`? 
`__name` was previously added to the stream tag tagset (using `NOIT_METRIC_TAGSET_STREAM`).
`__check_uuid` was previously added to the check tag tagset (using `NOIT_METRIC_TAGSET_CHECK`).
Do I need to separate these tags so that the fixup hook still uses those same values for `__name` and `__check_uuid`?

Reply:
> Does anyone know what noit_metric_tagset_fixup_hook_invoke does?

It does what all hooks do, it allows for an interception to alter the existing function